staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/conversion.go

```diff
@@ -65,6 +65,14 @@ func ToStructuralOpenAPIV2(in *structuralschema.Structural) *structuralschema.St
 				changed = true
 			}
 
+			if s.XPreserveUnknownFields {
+				// unknown fields break if items or properties are set in kubectl
+				s.Items = nil
+				s.Properties = nil
+
```

Comment:
AdditionalProperties?

Reply:
or is that one fine because kubectl will switch to the map case and then start fresh for the map values?

Reply:
no, kubectl doesn't have issues with unknown keys when additionalProperties are set

Reply:
We have an publishing kubectl e2e test, don't we? Can we add that there?

Reply:
added tests for it at the root and in a nested field